src/app/projects/states/dashboard/directives/student-task-list/student-task-list.tpl.html

```diff
@@ -7,7 +7,7 @@
       id="{{task.taskKeyToIdString()}}"
       ng-click="setSelectedTask(task)"
       ng-class="{selected: isSelectedTask(task)}"
-      ng-repeat="task in filteredTasks">
+      ng-repeat="task in filteredTasks | orderBy: 'topWeight'">
       <div class="task-data">
         <h4 class="list-group-item-heading">
           {{task.definition.name}}
```

Comment:
Not sure what we do with it, but it should be used somewhere.

Reply:
Do you want a tooltip like this when hovered?
![Screenshot from 2019-04-18 17-36-12](https://user-images.githubusercontent.com/29236609/56344374-96246380-6200-11e9-8853-e5d0d15a4fff.png)

Reply:
No, let’s put it at the top of the main pane - where the title of the task is above the task sheet or details. See if it looks ok there... otherwise maybe in the task details.

Reply:
I tried a few places. Have a look.

![Screenshot from 2019-04-19 12-37-27](https://user-images.githubusercontent.com/29236609/56402329-34b3d180-62a0-11e9-9ebe-6e6efdf14610.png)
![Screenshot from 2019-04-19 12-34-53](https://user-images.githubusercontent.com/29236609/56402330-354c6800-62a0-11e9-964f-6b09b66efe26.png)
![Screenshot from 2019-04-19 12-29-42](https://user-images.githubusercontent.com/29236609/56402331-354c6800-62a0-11e9-9bad-180a53013488.png)

Reply:
Probably not in the header - make it too big.

Where do you think it goes best? Could you also look at different styling for the different messages - make the message stand out when it’s getting closer.

Reply:
I think it looks best in the description card in the key information area. Let me know if you have anything else in mind.

Styling as in color? We can do something like this: 
 "Complete this grade task to get ahead" - We can show this in blue(text-primary in bootstrap) if the due date is after 3 days, yellow(text-warning) otherwise. 

![Screenshot from 2019-04-19 19-32-48](https://user-images.githubusercontent.com/29236609/56418517-79f4f500-62da-11e9-889b-eee701aa8b7c.png)

![Screenshot from 2019-04-19 19-29-31](https://user-images.githubusercontent.com/29236609/56418518-7b262200-62da-11e9-857e-5863516b163c.png)


And for another message "Complete this grade task to get back on track" we can show in red(text-danger) maybe?

![Screenshot from 2019-04-19 19-29-10](https://user-images.githubusercontent.com/29236609/56418541-84af8a00-62da-11e9-98fd-eeb453325dc4.png)

Reply:
Doesn't looks quite right here. Maybe add it in with the task status.

Reply:
Check it out.

![Screenshot from 2019-04-20 18-47-45](https://user-images.githubusercontent.com/29236609/56455064-48912d80-639d-11e9-9e43-39fed4efd44f.png)

![Screenshot from 2019-04-20 18-48-07](https://user-images.githubusercontent.com/29236609/56455065-48912d80-639d-11e9-82f9-11317f691b7e.png)

![Screenshot from 2019-04-20 18-48-31](https://user-images.githubusercontent.com/29236609/56455066-4929c400-639d-11e9-9d93-752b40292e46.png)

Reply:
It looks a little too big. Maybe use an alert. 

https://getbootstrap.com/docs/3.3/components/#alerts

Reply:
How about this?

![Screenshot from 2019-04-22 17-16-54](https://user-images.githubusercontent.com/29236609/56488774-b74bc380-6522-11e9-9918-61e4e6904244.png)

![Screenshot from 2019-04-22 17-17-05](https://user-images.githubusercontent.com/29236609/56488776-b74bc380-6522-11e9-96f7-76df93ab5374.png)

![Screenshot from 2019-04-22 17-17-13](https://user-images.githubusercontent.com/29236609/56488778-b7e45a00-6522-11e9-92e3-d3c86d73f7c2.png)

Reply:
Not sure we need to highlight it so much if it is a "get ahead" message. It could just be plain text. What is the difference between the two "get ahead" messages?

@jakerenzella thoughts?

Reply:
Right now I am using alerts but if you want then I can just make it normal text. 
The difference between 2 different messages is the due date. It the due date is less than 2 days then warning-alert otherwise just info-alert.

Reply:
Can someone clarify for me when this is being shown?

Reply:
It will be shown every time. This is related to sorting the tasks according to the top priority and assigning each task a 'top reason'. We are now displaying the top reason in the status card. So if a task has due date greater than 2 days then a message in blue is shown. If less than 2 days then message in yellow. If due date has passed then message in red.

Reply:
The get ahead message should probably only be there if it is well in advance of the first due date. Otherwise it is just "Work on this task now"

Reply:
The final changes are as follows:
To show the overdue alert if due date has passed.

![Screenshot from 2019-05-03 11-10-45](https://user-images.githubusercontent.com/29236609/57115790-5f1f8780-6d94-11e9-87e0-8f4f64865445.png)

To show the get ahead message if the due date is 2 weeks later. Else no message to show.

![Screenshot from 2019-05-03 11-11-06](https://user-images.githubusercontent.com/29236609/57115812-778fa200-6d94-11e9-8da9-6c2dfb5d920e.png)